documentation/ide-vscode-setup.html

```diff
@@ -0,0 +1,189 @@
+                            dedicated to downloads of Visual Studio Code and choose the package that matches your needs 
+                            and requirements.
+                            After Download install it, It will just take about a minute to install.
+                    </ul>    
+            </ul>
+        <p>
+            Start by Adding Java language support to VS Code by installing the popular minimum requirement  Java extensions.
+            There is a Java Extension Pack, which contains the most popular extensions for most Java developers: 
+        </p>
+        <p>
+            Launch your VS Code and bring up extensions view by clicking on the <b>Extensions</b> icon in the Activity Bar
+            on the left side of VS Code or  <b>View > Extensions</b> then write the name of extension <b>Java Extension Pack</b> and 
+            then click on install link to install all extensions for Pack.
+        </p>
+        <figure>
+            <a href="images/VSCode-1.png"><img src="images/VSCode-1.png" alt="Installing Java Extension Pack to VS Code."></a>
+	    <figcaption>Installing Java Extension Pack to VS Code.</figcaption>            
+        </figure>
+
+        <h2>Open the source code folder</h2>
+        <p>
+            The Openfire source code is using an Apache Maven project structure. As VS Code (with Maven Extension) recognizes this structure,
+            opening the source code into an VS Code project is very easy.
+        </p>
+        <p>
+            Select the 'Open' option -- <b>File > Open</b> and choose your Openfire source code folder then click on the open button. 
+            This will quickly import the Openfire source code into  VS Code .
+        </p>
+
+        <figure>
+            <a href="images/VSCode-2.png"><img src="images/VSCode-2.png" alt="Use the 'Open' functionality of VS Code."></a>
+            <a href="images/VSCode-3.png"><img src="images/VSCode-3.png" alt="Select the folder that will open in our VS Code."></a>
+            <a href="images/VSCode-4.png"><img src="images/VSCode-4.png" alt="Overview of project in Visual Studio Code"></a>
+	    <figcaption>Opening Openfire Source folder on VS Code.</figcaption>            
+        </figure>
+
+        <p>
+            VS Code will automatically recognize and process the Maven structure of Openfire (pom.xml file) and it will automatically download all dependencies.
+        </p>
+        <h2>Build Openfire from VS Code</h2>
+        <p>
+            To build the code straight from your IDE. The following instructions will explain how
+            you can build Openfire from the code that is in the VS Code workspace.
+        </p>
+        <p>
+            Select the 'New Terminal' option -- <b>Terminal > New Terminal</b> to open a terminal and run the command bellow to  build the complete project including plugins: 
+            <ul>
+                <span class="code"> mvn verify <br></span>
```

Comment:
I suggest we tell people to do `mvn clean verify` instead of `mvn verify`. The added `clean` instruction makes sure that the build is started from scratch, which will remove any potential issues from previous attempts (I have often wasted to much time on these).

Reply:
Certainly mention the `clean` option, but I frequently use just `mvn verify` so I can run Openfire direct from the `distributions/target/distribution-base` folder without having to re-configure every time ...

Reply:
For the intended audience, I'd make the instructions as fool-proof as possible. Maybe we could mention that you can leave off the 'clean' part if you don't want to loose configuration of Openfire.

Reply:
I think this can be resolved now